client/packages/coldchain/src/Equipment/ListView/AddFromScannerButton.tsx

```diff
@@ -30,19 +34,60 @@ export const AddFromScannerButtonComponent = () => {
   const equipmentRoute = RouteBuilder.create(AppRoute.Coldchain).addPart(
     AppRoute.Equipment
   );
-  const { mutateAsync: fetchAsset } = useAssets.document.fetch();
+  const { mutateAsync: scanAsset } = useAssets.document.scan();
+  const { mutateAsync: saveNewAsset } = useAssets.document.insert();
+  const { insertLog, invalidateQueries } = useAssets.log.insert();
+  const newAssetData = useRef<DraftAsset>();
+
+  const showCreateConfirmation = useConfirmationModal({
+    onConfirm: () => {
+      if (newAssetData.current) {
+        saveNewAsset(newAssetData.current)
+          .catch(e => error(t('error.unable-to-save-asset', { error: e }))())
```

Comment:
This catch doesn't work for some reason!
I tired adding `throw e` into the `useAssets.document.insert();` `useMutation:onError` but that didn't work.
Help please!

Reply:
It appears to work fine for me? Though you'd need to add `{{ error }}` to the 'error.unable-to-save-asset' translation for the error message to show... not entirely sure it should though!

The only time I can seem to see where it doesn't show a message is when `newAssetData.current` isn't defined.... I think somewhere we have an example of passing data into confirm (so we can pass directly from handleScanResult, and skip the ref thing completely) - but I can't find right now, and I know we're on a time crunch to get something for RC!

Can be a clean up for another PR, maybe as part of that other issue?

Reply:
I would move the catch below the .then() though! If `saveNewAsset` errors, we don't want to catch, and then try do the insertLog/redirect, should throw to below that